src/zenml/integrations/hyperai/orchestrators/hyperai_orchestrator.py

```diff
@@ -0,0 +1,445 @@
+                        self._generate_valid_path_format(mount_from),
+                        self._generate_valid_path_format(mount_to),
+                    )
+                    for mount_from, mount_to in step_settings.mounts_from_to.items()
+                ],
+            }
+
+            # Add dependency on upstream steps if applicable
+            upstream_steps = step.spec.upstream_steps
+            if isinstance(upstream_steps, list) and len(upstream_steps) > 0:
+                for upstream_step_name in upstream_steps:
+                    upstream_container_name = (
+                        f"{deployment_id}-{upstream_step_name}"
+                    )
+                    compose_definition["services"][container_name][
+                        "depends_on"
+                    ] = {
+                        upstream_container_name: {
+                            "condition": "service_completed_successfully"
+                        }
+                    }
+
+        # Convert into yaml
+        logger.info("Finalizing Docker Compose definition.")
+        compose_definition = yaml.dump(compose_definition)
+
+        # Connect to configured HyperAI instance
+        logger.info(
+            "Connecting to HyperAI instance and placing Docker Compose file."
+        )
+        paramiko_client: paramiko.SSHClient
+        if connector := self.get_connector():
+            paramiko_client = connector.connect()
+            if paramiko_client is None:
+                raise RuntimeError(
+                    "Expected to receive a `paramiko.SSHClient` object from the "
+                    "linked connector, but got `None`. This likely originates from "
+                    "a misconfigured service connector, typically caused by a wrong "
+                    "SSH key type being selected. Please check your "
+                    "`hyperai_orchestrator` configuration and make sure that the "
+                    "`ssh_key_type` of its connected service connector is set to the "
+                    "correct value."
+                )
+            elif not isinstance(paramiko_client, paramiko.SSHClient):
+                raise RuntimeError(
+                    f"Expected to receive a `paramiko.SSHClient` object from the "
+                    f"linked connector, but got type `{type(paramiko_client)}`."
+                )
+        else:
+            raise RuntimeError(
+                "You must link a HyperAI service connector to the orchestrator."
+            )
+
+        # Get container registry autologin setting
+        container_registry_autologin = self.config.container_registry_autologin
+        if container_registry_autologin:
+            logger.info(
+                "Attempting to automatically log in to container registry used by stack."
+            )
+
+            # Select stack container registry
+            container_registry = None
+            for component in stack.components.values():
```

Comment:
I think you can simply do `stack.container_registry` here as a shortcut

Reply:
Done